src/component/SortedLists.js

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import { FirestoreCollection, withFirestore } from 'react-firestore';
+import dayjs from 'dayjs';
+
+const SortedList = ({ token, handlePurchase, calculateIfPurchased }) => {
+  const concatPath = `/lists/${token}/items`;
+  const now = new Date();
+  const today = dayjs(now);
+
+  return (
+    <section className="listFrame">
+      <h1>&#128722;My Shopping List</h1>
+      {/* could also use &#128717; which is shopping bags */}
+      {/* // items that need to buy soon (fewer than seven days) */}
+      <FirestoreCollection
+        path={concatPath}
+        sort="numberOfDays:asc"
+        filter={['numberOfDays', '<', 7]}
+        render={({ isLoading, data }) => {
+          if (isLoading) {
+            return <div>Still Loading...</div>;
+          } else {
+            return (
+              <div id="soonItems">
+                <h2 className="itemsLabel">Soon Items</h2>
+                <ul>
+                  {data.map(item => (
+                    <li
+                      id={item.id}
+                      key={item.id}
+                      className={calculateIfPurchased(item)}
+                    >
+                      <div
+                        className={item.name}
+                        onClick={handlePurchase}
+                        id={item.id}
+                        aria-label="Soon Item"
+                        aria-required="true"
+                      >
+                        {item.name}
+                      </div>
+                    </li>
+                  ))}
+                </ul>
+              </div>
```

Comment:
I get the feeling that this chunk of code is the same as (or at least looks a lot like) the used inside the other `FirestoreCollection`s

What if you abstract it to its own component and save some LOC (Lines of Code)?

Reply:
I took a shot at this. Seems to work, saves about 40 lines of code, but possibly it could still be cleaner? The 4th section, inactive items, has some differences, so I left it the same.

Reply:
i debated doing this as well in the beginning and wondered how useful it would be @alejandronanez -- is there a general rule of thumb to go by for when to know to break stuff out into it's own component? should you just always do it